### ISSUE TITLE
co-teaching训练框架

### ISSUE THREAD

**Issue by PhilChina** [alibaba-damo-academy/fvlm#3]
hi, jianpeng

     @jianpengz  我看论文部分有提到使用了co-teaching这种框架，我印象中这种方式一般在处理噪声标签中使用，然后我看代码部分没有找到这部分的代码，所以想请教一下，co-teaching代码是否包含在这里还是目前没有release出来；目前只能看到这部分的代码，不确定是否为您说的co-teaching
`
class MomentumDistilationMixin:
    @torch.no_grad()
    def copy_params(self):
        for model_pair in self.model_pairs:
            if isinstance(model_pair[0], nn.Parameter):
                model_pair[1].data.copy_(model_pair[0].data)
                # model_pair[1].requires_grad = False
            else:
                for param, param_m in zip(
                    model_pair[0].parameters(), model_pair[1].parameters()
                ):
                    param_m.data.copy_(param.data)  # initialize
                    # param_m.requires_grad = False  # not update by gradient

    @torch.no_grad()
    def _momentum_update(self):
        for model_pair in self.model_pairs:
            for param, param_m in zip(
                model_pair[0].parameters(), model_pair[1].parameters()
            ):
                param_m.data = param_m.data * self.momentum + param.data * (
                    1.0 - self.momentum
                )
`

**Comment by jianpengz** [comment]
Yes, this is the implementation of a momentum network, and the co-teaching optimization could be referred to as Line 245-265 in blip_pretrain.py

**Comment by windygoo** [comment]
The co-teaching code is not included in this release as it needs the base models to produce accurate similarity estimation by training with large-scale data. However, the CT-RATE dataset is relatively small.

**Comment by DanyalMaq** [comment]
WIll the co-teaching code be released at a future point?
